### ISSUE TITLE
Enhancement request for `ToggleSwitch` to allow customization of "On" and "Off" labels

### ISSUE THREAD

**Issue by Swiftwork** [primer/react#2858]
### What I would like to see:

Add `statusLabels` property to `ToggleSwitch` component to customize "On" and "Off" labels. The reason being the words "On" and "Off" isn't always applicable. For instance with a label such as "Gate", "Opened/Closed" might be more reasonable, or with "Do you want emails?" maybe you would like "Yes/No".

### Screenshots

![image](https://user-images.githubusercontent.com/455178/204023531-00905205-1029-4fbc-8639-8b7bac5e0744.png)
![image](https://user-images.githubusercontent.com/455178/204023450-d6be021e-05ad-49c5-9fe7-d1f75d4e6b77.png)

### Proposed solution:
https://github.com/primer/react/pull/2616

Discussion can be held here as suggested by @broccolinisoup 

> Hi @Swiftwork 👋🏼 Thanks for the PR! For the contributions that introduce changes in the API, we recommend [creating an issue](https://github.com/primer/react/issues/new/choose) or [starting up a discussion](https://github.com/primer/react/discussions/new) to talk about the functionality, different approaches and confirm them with our design & accessibility design/engineering team to make a collective decision. I hope it is helpful!

- https://github.com/primer/react/pull/2616#pullrequestreview-1198551394

**Comment by tallys** [comment]
Hello! Thanks for opening this up. I'm adding this to our next Primer API Design working session agenda.

**Comment by mperrotti** [comment]
Thanks for your contribution @Swiftwork.

We discussed this in this week's working session, and we're not ready to add this functionality right now. We're concerned that consumers will mistakenly replace "On" and "Off" with text that labels the switch instead of text that communicates whether the switch is on or off.

The use-case you presented seems perfectly acceptable, but we haven't encountered any similar cases for GitHub yet. When and if we find a use-case like yours at GitHub, we'd be happy to reconsider this enhancement.

If you want to talk about workarounds to support your use-case, let me know and I'll share some ideas.

**Comment by Swiftwork** [comment]
@mperrotti That is unfortunate news. Can you please elaborate as to why this and the accompanying pull request won't be added? If it's the interface for specifying labels I am sure we can improve it. This is an important feature for our company and we are currently relying on patching the primer package which is not ideal. Not being able to change the labels seems like an oversight. Please do share what workaround exists to achieve the above images. Where on is replaced by opened and off is replaced by closed.

**Comment by tallys** [comment]
Hi @Swiftwork thanks for the follow up. At this time, we are following the guidance of our accessibility design team and prioritizing keeping the component as is as guardrails to enforce the accessible use. If we revisit this at another time, we'll pick this back up.

**Comment by Swiftwork** [comment]
@tallys Thanks for replying. However, I don't quite follow how restricting the wording of the toggle switch to **on** and **off** rather than allowing developers to choose something more contextually appropriate to their setting such as **open** and **close** for a gate is considered better for accessibility. If you could please elaborate on the reasoning.

@mperrotti Any suggestions on what workarounds I could employ?

**Comment by ericwbailey** [comment]
Hi there, @Swiftwork. I'm part of the accessibility design team and was tapped to chime in. 

You've pretty much hit the nail on the head of why we made this decision with this comment:

>allowing developers to choose something more contextually appropriate to their setting

The issue is that when you allow custom language for a switch control in this fashion you need to ensure that the language it uses makes sense for someone using assistive technology who may not be able to perceive its visible state. 

In terms of stewardship, that assumes that every person utilizing ToggleSwitch will make an informed decision about the language they use, which increases the risk of an inaccessible treatment. As Primer is designed to accommodate all of GitHub, this architectural choice helps us to ensure accessible experiences at scale, without having to manually review every ToggleSwitch instance.

In addition, I'd like to cite [this part of the ARIA spec](https://w3c.github.io/aria/#switch) (emphasis mine):

>A type of checkbox that represents <strong>on/off</strong> values

While the spec does not explicitly call for using "on" or "off", toggle switches are oftentimes misused, where another component would be more applicable for the design. Intentionally reducing the surface area via this restriction helps to indirectly reinforce that it is being used appropriately. 

Localization concerns also apply here, but are outside the scope of your immediate issue.

tl;dr: We made an intentional call governance-wise to reduce the surface area of potential accessibility issues from a frequently misused component.
